crates/rpc/rpc/src/txpool.rs

```diff
@@ -19,34 +20,37 @@ use tracing::trace;
 ///
 /// This type provides the functionality for handling `txpool` related requests.
 #[derive(Clone)]
-pub struct TxPoolApi<Pool, Eth> {
+pub struct TxPoolApi<N, Pool, Eth> {
     /// An interface to interact with the pool
     pool: Pool,
     tx_resp_builder: Eth,
+    _phantom: PhantomData<N>,
 }
```

Comment:
if we have primitives be an associated type on `TransactionCompat` instead, we can avoid this phantom data. I would suggest addressing this issue as 2 parts. the first, just adding the the primitives associated type to `TransactionCompat` and removing the generic `T`. the next step then moves the functions into trait. will open sub issue.